js/ui/data_grid/ui.data_grid.focus.js

```diff
@@ -11,7 +11,40 @@ var MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 9007199254740991/* IE11 */;
 gridCore.registerModule("focus", extend(true, {}, focusModule, {
     extenders: {
         controllers: {
+            focus: {
+                isFocusedRowInsideGroup: function(groupInfo) {
+                    var dataController = this.getController("data"),
+                        focusedRowKey = this.option("focusedRowKey"),
+                        rowIndex = dataController.getRowIndexByKey(focusedRowKey),
+                        group = dataController.getDataSource().group(),
+                        row = rowIndex >= 0 && dataController.getVisibleRows()[rowIndex],
+                        levelPath;
+
+                    if(row) {
+                        for(var i = 0; i < group.length; ++i) {
+                            levelPath = group[i].selector;
+                            if(groupInfo.path[i] !== row.data[levelPath]) {
+                                return false;
+                            }
+                        }
+                    }
+                    return true;
+                }
+            },
             data: {
+                changeRowExpand: function(path) {
+                    if(this.option("focusedRowEnabled")) {
+                        var groupInfo = this._dataSource._grouping.findGroupInfo(path),
+                            focusController = this.getController("focus"),
+                            needFocusGroupRow = groupInfo.isExpanded && focusController.isFocusedRowInsideGroup(groupInfo);
```

Comment:
Use isRowExpanded public method instead of internal _dataSource._grouping.findGroupInfo and groupInfo.isExpanded